frontend/app/src/types/room.ts

```diff
@@ -0,0 +1,6 @@
+import { DifficultyType, TopicTagType } from "@/constants/question";
+
+export type Preferences = {
+  difficulty: DifficultyType[] | undefined;
```

Comment:
may i know why you let the attributes be possibly `undefined` as well? if its because of the form state in `SelectPreferencesCard`, you could initialize the form with the default state as empty arrays

https://react-hook-form.com/docs/useform#defaultValues

Reply:
resolved in [8a6bb84](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/13/commits/8a6bb8476828080ee5a7b966cfd70c012d05401d)